File: 33-search-in-rotated-sorted-array/search-in-rotated-sorted-array.c
int search(int* nums, int numsSize, int target) {
    int st=0;
    int end = numsSize-1;
    while(st<=end){
        int mid=(st+(end-st)/2);
        if (nums[mid]==target){
            return mid;
        }
        else if(nums[st]<=nums[mid]){
            if(nums[st]<=target && target<=nums[mid]){
                end=mid-1;
            }
            else{
                st=mid+1;
            }
        }
        else{
            if(nums[mid]<=target && target<=nums[end]){
                st=mid+1;
            }
            else{
                end=mid-1;
            }
        }

    }
    return -1;
}

File: 33-search-in-rotated-sorted-array/search-in-rotated-sorted-array.py
class Solution(object):
    def search(self,nums, target):
        st = 0
        end = len(nums) - 1
        
        while st <= end:
            mid = st + (end - st) // 2
            
            if nums[mid] == target:
                return mid
            elif nums[st] <= nums[mid]:
                if nums[st] <= target <= nums[mid]:
                    end = mid - 1
                else:
                    st = mid + 1
            else:
                if nums[mid] <= target <= nums[end]:
                    st = mid + 1
                else:
                    end = mid - 1
        
        return -1
